relay.go

```diff
@@ -230,23 +238,23 @@ func NewRelayer(ch *Channel, conn *Connection) *Relayer {
 }
 
 // Relay is called for each frame that is read on the connection.
-func (r *Relayer) Relay(f *Frame) error {
+func (r *Relayer) Relay(f *Frame) (bool, error) {
 	if f.messageType() != messageTypeCallReq {
 		err := r.handleNonCallReq(f)
 		if err == errUnknownID {
 			// This ID may be owned by an outgoing call, so check the outbound
 			// message exchange, and if it succeeds, then the frame has been
 			// handled successfully.
 			if err := r.conn.outbound.forwardPeerFrame(f); err == nil {
-				return nil
+				return false, nil
```

Comment:
for each `false`, add a single line of comment about why we don't need to release the frame

or alternatively, 
```
// Comment about why we don't need to release the frame in these paths
const relayNoRelease = false
```